bundles/org.openhab.core.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/config/EnrichedConfigDescriptionDTOMapper.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.io.rest.core.config;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.config.core.ConfigDescription;
+import org.eclipse.smarthome.config.core.ConfigDescriptionParameter;
+import org.eclipse.smarthome.config.core.dto.ConfigDescriptionDTO;
+import org.eclipse.smarthome.config.core.dto.ConfigDescriptionDTOMapper;
+import org.eclipse.smarthome.config.core.dto.ConfigDescriptionParameterDTO;
+import org.eclipse.smarthome.config.core.dto.ConfigDescriptionParameterGroupDTO;
+
+/**
+ * The {@link EnrichedConfigDescriptionDTOMapper} is a utility class to map config descriptions into config
+ * descriptions data transform objects (DTOs) containing enriched config descriptions parameters.
+ *
+ * @author Christoph Weitkamp - Initial contribution
+ */
+@NonNullByDefault
+public class EnrichedConfigDescriptionDTOMapper extends ConfigDescriptionDTOMapper {
+
+    /**
+     * Maps configuration description into configuration description DTO object.
+     *
+     * @param configDescription the configuration description (not null)
+     * @return enriched configuration description DTO object
+     */
+    public static ConfigDescriptionDTO map(ConfigDescription configDescription) {
+        List<ConfigDescriptionParameterGroupDTO> parameterGroups = mapParameterGroups(
+                configDescription.getParameterGroups());
+        List<ConfigDescriptionParameterDTO> parameters = mapEnrichedParameters(configDescription.getParameters());
+        return new ConfigDescriptionDTO(toDecodedString(configDescription.getUID()), parameters, parameterGroups);
+    }
+
+    public static List<ConfigDescriptionParameterDTO> mapEnrichedParameters(
```

Comment:
Shouldn't this be a private method? If not, JavaDoc is missing 😉. 
```suggestion
    private static List<ConfigDescriptionParameterDTO> mapEnrichedParameters(
```

Reply:
I added JavaDoc as the `mapParameters()` method of the base class is public too.